### PR TITLE
Adjust failing Windows tests to make them pass

### PR DESCRIPTION
**Pull request by c-poole**
We had another instance of a dtype that doesn't exist in windows numpy showing up in some tests. Modified to skip in Windows.

### CODE REVIEW

**Comment by vtomole** [comment]
What i did was delete the non-supporting types instead. See: https://github.com/quantumlib/Cirq/pull/1540/files#diff-15443b8109d9fb57ef919d410ae5f5d0R53

File: cirq/protocols/approximate_equality_test.py [hunk 1]
```diff
@@ -38,6 +40,9 @@ def test_approx_eq_mixed_primitives():
     assert not cirq.approx_eq(1, 1.0 + 1e-10, atol=1e-11)
 
 
+@pytest.mark.skipif(sys.platform.startswith('win'),
```
Comment (vtomole) [inline cirq/protocols/approximate_equality_test.py]: Let's skip if `float128` or `complex256` is not supported like https://github.com/quantumlib/Cirq/blob/0452fd817fd6a52e23682712484a8a5c047d30d0/cirq/google/sim/xmon_simulator_test.py#L375

File: cirq/protocols/approximate_equality_test.py [hunk 2]
```diff
@@ -73,6 +78,9 @@ def test_decimal_compatibility():
     assert not cirq.approx_eq(Decimal('NaN'), Decimal('-Infinity'), atol=1e-9)
 
 
+@pytest.mark.skipif(sys.platform.startswith('win'),
```
Comment (vtomole) [inline cirq/protocols/approximate_equality_test.py]: skipif for types like my previous comment.

File: cirq/protocols/approximate_equality_test.py (hunk 1 above)
Comment (c-poole) [reply cirq/protocols/approximate_equality_test.py]: Adjusted skip conditions in both adjusted tests.

**Comment by c-poole** [comment]
Whenever a solution for this gets merged, all tests outside of contrib pass on my Windows machine. There are two circuit diagram comparison tests in contrib that are failing for reasons I suspect are similar to the ion_device tests.

**Comment by c-poole** [comment]
I modified this change so that instead of skipping the tests if there is a bad data type inside them, the test only utilizes the bad data types if the machine recognizes them. This seems like the best route to take for this. Ready for re-review @vtomole.

**Pull request merged**